plugins/geotiff/main.py

```diff
@@ -0,0 +1,97 @@
+import json
+import os
+
+import rasterio
+from pyproj import Proj
+
+from vistas.core.gis.extent import Extent
+from vistas.core.plugins.data import RasterDataPlugin, VariableStats
+
+
+class GeoTIFF(RasterDataPlugin):
+
+    id = 'geotiff'
+    name = 'GeoTIFF Data Plugin'
+    description = 'Reads GeoTIFF (.tif) files'
+    author = 'Conservation Biology Institute'
+    version = '1.0'
+    extensions = [('tif', 'GeoTIFF')]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._name = None
+        self._shape = None
+        self._extent = None
+        self._resolution = None
+        self._nodata = None
+        self._count = None
+
+    def load_data(self):
+        self._name = self.path.split(os.sep)[-1].split('.tif')[0]
+        with rasterio.open(self.path) as src:
+            projection = Proj(src.crs)
+            self._shape = src.shape
+            self._resolution = src.res[0]
+            xmin, ymin = src.xy(src.height, 0, 'ul')
+            xmax, ymax = src.xy(0, src.width, 'ul')
+            self._extent = Extent(xmin, ymin, xmax, ymax, projection)
+            self._nodata = src.nodata
+            self._count = src.count
+
+    @property
+    def data_name(self):
```

Comment:
The way that class variables and instance variables overlap can be a little confusing at first. Essentially, referencing `self.` will use the instance variable, if set, else the class variable. However, assigning to `self.` will always set the instance variable (the property method is not a variable and so a special case). Consider the following:

```python
class Foo:
    x = 5

# Changing the instance variable doesn't affect the class variable
>>> f = Foo()
>>> f.x
5
>>> Foo.x
5
>>> f.x = 7
>>> f.x
7
>>> Foo.x
5

# Likewise, changing the class variable after the instance is set has no affect
>>> Foo.x = 3
>>> f.x
7

# But before an instance variable is set, changing the class variable will change f.x
>>> f = Foo()
>>> f.x
3
>>> Foo.x = 11
>>> f.x
11

# Extra caution is required when using mutable objects at the class level
class Bar:
    d = {}

>>> b1 = Bar()
>>> b1.d['foo'] = 'Bar'
>>> b2 = Bar()
>>> b2.d['bar'] = 'Foo'
>>> b2.d
{'foo': 'Bar', 'bar': 'Foo'}
>>> b1.d is b2.d
True
>>> b2.d is Bar.d
True
```

Since the parent class here defines `data_name` as a property, setting `self.data_name = ` is actually going to attempt to lookup a setter. None is defined, hence the error.

> It seems a little unintuitive to override a property with None in the class declaration and then set it up as an instance variable. It works here but doesn't feel quite right to me.
Think of it this way: the default behavior is for `data_name` to be a property method. The `GeoTIFF` class is overriding that default to turn it into a variable instead with `data_name = None` at the class level. That said, I'm open to approaching it the other way: make `data_name` et. al. variables in the parent class and let plugins override then as property methods as needed.